app/services/scene-collections/index.ts

```diff
@@ -0,0 +1,887 @@
+import { StatefulService, mutation } from 'services/stateful-service';
+import { Inject } from 'util/injector';
+import { SceneCollectionsServerApiService } from 'services/scene-collections/server-api';
+import Vue from 'vue';
+import { RootNode } from './nodes/root';
+import { SourcesNode, ISourceInfo } from './nodes/sources';
+import { ScenesNode, ISceneSchema } from './nodes/scenes';
+import { SceneItemsNode, ISceneItemInfo } from './nodes/scene-items';
+import { TransitionNode } from './nodes/transition';
+import { HotkeysNode } from './nodes/hotkeys';
+import path from 'path';
+import electron from 'electron';
+import fs from 'fs';
+import { parse } from './parse';
+import { ScenesService } from 'services/scenes';
+import { SourcesService } from 'services/sources';
+import { E_AUDIO_CHANNELS } from 'services/audio';
+import { AppService } from 'services/app';
+import { HotkeysService } from 'services/hotkeys';
+import namingHelpers from '../../util/NamingHelpers';
+import { WindowsService } from 'services/windows';
+import { UserService } from 'services/user';
+import { OverlaysPersistenceService, IDownloadProgress } from './overlays';
+
+const uuid = window['require']('uuid/v4');
+
+interface ISceneCollectionsManifestEntry {
+  id: string;
+  name: string;
+  serverId?: number;
+  deleted: boolean;
+  modified: string;
+}
+
+interface ISceneCollectionsManifest {
+  activeId: string;
+  collections: ISceneCollectionsManifestEntry[];
+}
+
+export const NODE_TYPES = {
+  RootNode,
+  SourcesNode,
+  ScenesNode,
+  SceneItemsNode,
+  TransitionNode,
+  HotkeysNode
+};
+
+const DEFAULT_COLLECTION_NAME = 'Scenes';
+
+export interface ISceneCollectionSchema {
+  name: string;
+  id: string;
+
+  scenes: {
+    id: string;
+    name: string;
+    sceneItems: { sceneItemId: string, sourceId: string }[]
+  }[];
+
+  sources: {
+    sourceId: string;
+    name: string;
+    type: string;
+    channel: number;
+  }[];
+}
+
+/**
+ * V2 of the scene collections service:
+ * - Completely asynchronous
+ * - Server side backup
+ */
+export class SceneCollectionsService extends StatefulService<ISceneCollectionsManifest> {
```

Comment:
We have to expose some API for Streamdeck. To do that create a file with `#serviceName#-api.ts` name and define an interface SceneCollectionsServiceApi. List the methods to expose and add some jsdoc for it. To have a typecheck add `implements SceneCollectionsServiceApi` here.